### PR TITLE
Update ruby, favicon, configuration

### PR DESCRIPTION
**Pull request by davidwessman**
- Ruby 2.4.0
- Rails 4.2.8
- Changing configurations to be up to date with 4.2
- Use procfile to run migrations, removing multi-buildpack

### CODE REVIEW

**Comment by navro** [comment]
Sorry, forgot to comment on this

File: config/initializers/filter_parameter_logging.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password,:password_confirmation,:civic]
```
Comment (navro) [inline config/initializers/filter_parameter_logging.rb]: Should `:password_confirmation` be unfiltered?

File: config/environments/production.rb [hunk 2]
```diff
@@ -39,11 +40,7 @@
   config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for nginx
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
-  if ENV.fetch('FORCE_SSL').to_s.downcase == 'true'
-    config.force_ssl = true
-  else
-    config.force_ssl = false
-  end
+  config.force_ssl = ENV.fetch('FORCE_SSL').to_s.casecmp('true')
```
Comment (navro) [inline config/environments/production.rb]: Nice :)

File: config/environments/production.rb [hunk 3]
```diff
@@ -100,7 +97,4 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
-  PUBLIC_URL = ENV['APPLICATION_URL']
-
-  config.secret_key_base = ENV['SECRET_KEY_BASE']
```
Comment (navro) [inline config/environments/production.rb]: Have we moved this somewhere else?

Comment (davidwessman) [inline config/environments/production.rb]: Standard for > Rails 4.2 to have it in `config/secrets.yml`.
Was defined two times.

**Pull request merged**